src/main/java/frc/robot/subsystems/intake/Intake.java

```diff
@@ -0,0 +1,157 @@
+     */
+    public double getLastControlledTime() {
+        if (lastControlledTime.isEmpty()) {
+            return Constants.PERIOD_TIME_MILLISECONDS;
+        }
+        return lastControlledTime.get();
+    }
+
+    /** Returns the cosine of the arm angle in degrees off of the horizontal. */
+    public double getCosineArmAngle() {
+        return Math.cos(
+                Units.degreesToRadians(
+                        pivotEncoder.getPosition() - IntakeConstants.INTAKE_POSITION_WHEN_HORIZONTAL_DEGREES));
+    }
+
+    public double getOffsetAbsPositionDeg() {
+        return pivotAbsoluteEncoder.getPosition() + IntakeCal.INTAKE_ABSOLUTE_ENCODER_ZERO_OFFSET_DEG;
+      }
+
+    public void moveToPos(IntakePosition pos) {
+        pivotController.setGoal(intakePositionMap.get(pos));
+        double intakeDemandVoltsA = pivotController.calculate(getOffsetAbsPositionDeg());
+    
+        double intakeDemandVoltsC = IntakeCal.ARBITRARY_INTAKE_PIVOT_FEEDFORWARD_VOLTS * getCosineArmAngle();
+
+        double currentTimestamp = Timer.getFPGATimestamp();
+        double timestampDifference = currentTimestamp - getLastControlledTime();
+        lastControlledTime = Optional.of(currentTimestamp);
+        double intakeDemandVoltsB = IntakeCal.INTAKE_PIVOT_FEEDFORWARD.calculate(pivotController.getSetpoint().velocity,
+                pivotController.getGoal().velocity, timestampDifference);
+
+        pivotMotor.setVoltage(intakeDemandVoltsA + intakeDemandVoltsB + intakeDemandVoltsC);
+    }
+
+    public boolean atDesiredIntakePosition() {
+        double desiredPositionDegrees = intakePositionMap.get(desiredPosition);
+        double intakePositionDegrees = pivotEncoder.getPosition();
+        return Math.abs(intakePositionDegrees - desiredPositionDegrees) <= IntakeCal.INTAKE_MARGIN_DEGREES;
+    }
+
+    public boolean atIntakePosition(IntakePosition pos) {
+        double checkPositionDegrees = intakePositionMap.get(pos);
+        double intakePositionDegrees = pivotEncoder.getPosition();
+        return Math.abs(intakePositionDegrees - checkPositionDegrees) <= IntakeCal.INTAKE_MARGIN_DEGREES;
+    }
+
+    /**
+     * 90 degrees represents when the intake is at the shooter
+     * therefore, being clear of the conveyor zone would be greater than the
+     * threshold
+     */
+    public boolean clearOfConveyorZone() {
+        return pivotEncoder.getPosition() > IntakeCal.CONVEYOR_ZONE_THRESHOLD_DEGREES;
+    }
+
+    public void startRollers() {
+        intakeTalonFront.set(IntakeCal.INTAKING_POWER);
+    }
+
+    public void stopRollers() {
+        intakeTalonFront.stopMotor();
+    }
+
+    public void reverseRollers() {
+        intakeTalonFront.set(IntakeCal.REVERSE_INTAKING_POWER);
+    }
+
+    public void periodic() {
+        if (atIntakePosition(IntakePosition.DEPLOYED)) {
+            startRollers();
+        } else {
+            stopRollers();
+        }
```

Comment:
I think ultimately this won't be correct. We will have the intake out when we climb. So probably we should make commands explicitly call start and stop themselves.

Reply:
so should I just delete this?